src/jmc/compile/command/_flow_control.py

```diff
@@ -340,10 +340,14 @@ def switch(command: list[Token], datapack: DataPack,
     # Parse variable
     tokens = tokenizer.parse(
         command[1].string[1:-1], command[1].line, command[1].col + 1, expect_semicolon=False)[0]
+    
     if len(tokens) > 1:
-        raise JMCSyntaxException(
-            f"Unexpected token({tokens[1].string})", tokens[1], tokenizer)
-
+        try: 
+            tokens = [merge_obj_selector(tokens, tokenizer, datapack)]
```

Comment:
Use `is_obj_selector` (or whatever its name is, I forgot) function instead of try except

Reply:
Aight, I made the change!